### ISSUE TITLE
Eliminate optional setting of nodes as Internal Pointers

### ISSUE THREAD

**Issue by PushkarBettadpur** [eclipse/omr#4891]
At present, transformations to mark nodes as internal pointers are guarded in `setIsInternalPointer` with the following:

```
performNodeTransformation2(c, "O^O NODE FLAGS: Setting internalPointer flag on node %p to %d\n", self(), v)
```

While the return value of this guard is determined by developer controlled options/flags such as `lastOptSubIndex, lastOptTransformationIndex, TR_TraceNodeFlags, TR_TraceOptDetails and TR_CountOptTransformations`, based on our current knowledge, it is hard to envision a scenario wherein the program would remain functionally correct after calling `setIsInternalPointer` and failing to set the flag as desired.

This issue is to address and track the (likely) elimination of this guard.

**Comment by andrewcraik** [comment]
Agreed that this shouldn't be a perform transformation (though the code calling to create it should have the overall transform guarded properly). This should become a dumpOptDetails line to report that the node is set as an internal pointer in the log to ease debugging.
